### PR TITLE
Release v0.0.7

### PR DESCRIPTION
**Pull request by tristan-deep**
version bump to v0.0.7

Release message below:

---
Many smaller improvements to the pipeline, beamforming, parameter dependencies, and documentation.
Also fixed the import bug for missing package tkinter and more intuitive import error handling for missing backend installations. All docstrings are now tested with Sphinx doctest.

Some highlighted features:
- Decomposed Diffusion Sampling in #116 
- Added waveforms and TGC to scan in https://github.com/tue-bmd/zea/pull/125
- A few funcs were relocated to more appropriate modules, see https://github.com/tue-bmd/zea/pull/126
- `zea.tensor_ops.vmap` in https://github.com/tue-bmd/zea/pull/144

> [!IMPORTANT]  
> Starting from zea v0.0.7, the minimum required Keras version is >=3.12

## What's Changed
* Import(ant) fix (tkinter + backend missing) by @tristan-deep in https://github.com/tue-bmd/zea/pull/115
* Decomposed Diffusion Sampling Implementation by @OisinNolan in https://github.com/tue-bmd/zea/pull/116
* Simplified phase rotation and improved documentation  by @vincentvdschaft in https://github.com/tue-bmd/zea/pull/123
* Pipeline traces outputs of Operations to see which keys are actually needed at the input by @wesselvannierop in https://github.com/tue-bmd/zea/pull/117
* Fix pipeline timing, rename fnum & angles and posterior_sampling more generic by @wesselvannierop in https://github.com/tue-bmd/zea/pull/118
* Fix dependency error with selected_transmits in Scan object by @vincentvdschaft in https://github.com/tue-bmd/zea/pull/128
* Improved documentation for transmit indexing in load_file by @vincentvdschaft in https://github.com/tue-bmd/zea/pull/124
* Fix Scan pickling and equality comparison after pickling by @wesselvannierop in https://github.com/tue-bmd/zea/pull/121
* Adding waveforms and TGC to scan, changing waveform saving by @vincentvdschaft in https://github.com/tue-bmd/zea/pull/125
* Feature time to peak by @vincentvdschaft in https://github.com/tue-bmd/zea/pull/131
* Refactor utilities: migrate save_to_gif to zea.io_lib, translate to zea.ops by @tristan-deep in https://github.com/tue-bmd/zea/pull/126
* Attributes of Parameters can be updated in-place now by @wesselvannierop in https://github.com/tue-bmd/zea/pull/133
* Map coordinates func dtype fix for higher order interpolation by @tristan-deep in https://github.com/tue-bmd/zea/pull/136
* Update papers list on about page and reformat BibTeX by @tristan-deep in https://github.com/tue-bmd/zea/pull/135
* Plot frustum vertices extended visualization options and improved tests by @tristan-deep in https://github.com/tue-bmd/zea/pull/137
* Remove deprecated firebase docs hosting files by @benluijten in https://github.com/tue-bmd/zea/pull/141
* Manually specify output keys of Pipeline for more robust tracing by @wesselvannierop in https://github.com/tue-bmd/zea/pull/139
* Added Sphinx doctests by @tristan-deep in https://github.com/tue-bmd/zea/pull/142
* Adds functional ops `envelope_detect`, `normalize` and `log_compress` by @wesselvannierop in https://github.com/tue-bmd/zea/pull/140
* Refactor mapping functions into `zea.tensor_ops.vmap` by @wesselvannierop in https://github.com/tue-bmd/zea/pull/144
* Update model documentation by @tristan-deep in https://github.com/tue-bmd/zea/pull/148
* Split the examples toc into multiple levels by @Louisvh in https://github.com/tue-bmd/zea/pull/149

## Summary by CodeRabbit

* **Chores**
  * Project version and release date bumped to 0.0.7.

* **New Features**
  * Added wrappers for several Keras operations to improve runtime compatibility.

* **Bug Fixes / API**
  * Updated padding and resizing data-layer bindings—public inheritance/behavior changed (may affect integrations).
  * Adjusted async scheduling logic for movie playback to a different loop-detection approach.

* **Tests**
  * Tightened/adjusted numeric assertions and formatting in tensor operation tests.

### CODE REVIEW

**Review by wesselvannierop — APPROVED** [review]

**Comment by tristan-deep** [comment]
Internally, `keras.src.layers.preprocessing.tf_data_layer.TFDataLayer` was updated to `keras.src.layers.preprocessing.data_layer.DataLayer`, see https://github.com/keras-team/keras/pull/21598.

I now updated this and also pinned keras to >=3.12.

**Comment by wesselvannierop** [comment]
@tristan-deep I think the tests that are failing should be fixed by setting a fixed seed for the inputs. Also, there are many more tests that have `np.random` inside the `pytest.mark.parametrize` so we might want to do a single PR that fixes all those.

**Comment by tristan-deep** [comment]
> @tristan-deep I think the tests that are failing should be fixed by setting a fixed seed for the inputs. Also, there are many more tests that have `np.random` inside the `pytest.mark.parametrize` so we might want to do a single PR that fixes all those.

That is probably better indeed. And then some global precision parameter as well (some tests can deviate from that).
